### PR TITLE
Define shared theme variables once

### PR DESCRIPTION
**Pull request by rhysd**
These variables are defined in both light mode and dark mode. So they can be defined unconditionally.

```css
--fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace;
--base-text-weight-semibold: 600;
--base-text-weight-medium: 500;
--base-text-weight-normal: 400;
--base-size-16: 1rem;
--base-size-8: 0.5rem;
--base-size-4: 0.25rem;
```

This PR defines these variables once instead of defining them in respective modes so that we can reduce the size of output.

Here is the diff of the output before/after applying this PR:

```diff
--- test.css	2024-06-15 09:53:59.169379660 +0900
+++ test2.css	2024-06-15 10:08:38.183798342 +0900
@@ -1,15 +1,18 @@
+.markdown-body {
+  --base-size-4: 0.25rem;
+  --base-size-8: 0.5rem;
+  --base-size-16: 1rem;
+  --base-text-weight-normal: 400;
+  --base-text-weight-medium: 500;
+  --base-text-weight-semibold: 600;
+  --fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace;
+}
+
 @media (prefers-color-scheme: dark) {
   .markdown-body,
   [data-theme="dark"] {
     /*dark*/
     color-scheme: dark;
-    --fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace;
-    --base-text-weight-semibold: 600;
-    --base-text-weight-medium: 500;
-    --base-text-weight-normal: 400;
-    --base-size-16: 1rem;
-    --base-size-8: 0.5rem;
-    --base-size-4: 0.25rem;
     --focus-outlineColor: #1f6feb;
     --fgColor-default: #e6edf3;
     --fgColor-muted: #8d96a0;
@@ -68,13 +71,6 @@
   [data-theme="light"] {
     /*light*/
     color-scheme: light;
-    --fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace;
-    --base-text-weight-semibold: 600;
-    --base-text-weight-medium: 500;
-    --base-text-weight-normal: 400;
-    --base-size-16: 1rem;
-    --base-size-8: 0.5rem;
-    --base-size-4: 0.25rem;
     --focus-outlineColor: #0969da;
     --fgColor-default: #1f2328;
     --fgColor-muted: #636c76;
```

### CODE REVIEW

**Comment by hyrious** [comment]
Would you have a test on one-theme-only styles? Because `applyStyles()` need to read all variables (including the shared ones) to replace them. You can follow these steps:

```bash
node test.js
wget https://cdn.jsdelivr.net/npm/github-markdown-css/github-markdown-dark.css
git diff --no-index github-markdown-dark.css dist/dark.css
```

**Comment by rhysd** [comment]
I see. The shared variables are not applied on generating the style. I didn't consider it.

**Comment by rhysd** [comment]
5d1531a fixed the issue.

```diff
diff --git a/github-markdown-dark.css b/dist/dark.css
index d1a68e7..41c8f72 100644
--- a/github-markdown-dark.css
+++ b/dist/dark.css
@@ -1,3 +1,13 @@
+.markdown-body {
+  --base-size-4: 0.25rem;
+  --base-size-8: 0.5rem;
+  --base-size-16: 1rem;
+  --base-text-weight-normal: 400;
+  --base-text-weight-medium: 500;
+  --base-text-weight-semibold: 600;
+  --fontStack-monospace: ui-monospace, SFMono-Regular, SF Mono, Menlo, Consolas, Liberation Mono, monospace;
+}
+
 /*dark*/
 
 .markdown-body {
@@ -1100,4 +1110,4 @@
 
 .markdown-body>*:first-child>.heading-element:first-child {
   margin-top: 0 !important;
-}
+}
\ No newline at end of file
```

File: index.js [hunk 1]
```diff
@@ -531,6 +523,12 @@ export default async function getCSS({
 				}],
 			});
 		}
+
+		rules.unshift({
+			type: 'rule',
+			selectors: ['.markdown-body'],
+			declarations: sharedDeclarations,
+		});
```
Comment (hyrious) [inline index.js:531]: I don't think one-theme-only styles needs these variables.

```suggestion

			rules.unshift({
				type: 'rule',
				selectors: ['.markdown-body'],
				declarations: sharedDeclarations,
			});
		}
```

**Review by hyrious — APPROVED** [review]

File: index.js (hunk 1 above)
Comment (hyrious) [reply index.js:531]: However, if moved like above, `dist/vars/dark.css` will miss these variables.

**Review by hyrious — APPROVED** [review]

**Pull request merged**